apis/r/R/datasets.R

```diff
@@ -0,0 +1,87 @@
+#' SOMA Example Datasets
+#'
+#' @description
+#' Access example SOMA objects bundled with the tiledbsoma package.
+#'
+#' Use `list_datasets()` to list the available datasets and `load_dataset()` to
+#' load a dataset into memory using the appropriate SOMA class. The
+#'  `extract_dataset()` method returns the path to the extracted dataset without
+#'  loading it into memory.
+#'
+#' @details
+#' The SOMA objects are stored as `tar.gz` files in the package's `extdata`
+#' directory. Calling `load_dataset()` extracts the `tar.gz` file to the
+#' specified `dir`, inspects its metadata to determine the appropriate SOMA
+#' class to instantiate, and returns the SOMA object.
+#'
+#' @examples
+#' soma_pbmc_small <- load_dataset("soma-exp-pbmc-small")
+#'
+#' @name example-datasets
+NULL
+
+#' @rdname example-datasets
+#' @return
+#'  - `list_datasets()` returns a character vector of the available datasets.
+#' @importFrom tools file_path_sans_ext
+#' @export
+list_datasets <- function() {
+  data_dir <- example_data_dir()
+  stopifnot("The dataset directory does not exist" = dir.exists(data_dir))
```

Comment:
This line is unneeded as `system.file(mustWork = TRUE)` will throw an error if the directory doesn't exist

Reply:
Good catch. I thought I had removed this already after your earlier hint.